### ISSUE TITLE
Parent page dropdown: screen reader issues

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#27385]
**Describe the bug**

When navigating to the parent page dropdown under "Page Attributes" in the sidebar, VoiceOver on Safari does not announce the input label, but goes straight to announcing number of options. 

Additionally, if a selection already exists, it is not announced either with VoiceOver/Safari or with NVDA/Firefox, so the user has no way of knowing if an option is currently selected. 

**To reproduce**
1. Go to any page (there must be multiple pages on the website);
2. Using VoiceOver, navigate to Page Attributes in the sidebar, then tab once to land in the parent page dropdown;
3. Verify that VoiceOver doesn't announce the component name (which is "Parent page);
4. Select an option from the dropdown;
5. Tab away from the dropdown, and then tab back into it;
6. Verify that the previously selected option is not announced.

**Expected behavior**

Expected both component label, and existing selection (if there is one), to be read out by screen reader when focusing on the input.

**Editor version (please complete the following information):**
- WordPress version: 5.6 RC1
- If the Gutenberg plugin is installed, which version is it? 9.4.1

**Comment by alexstine** [comment]
@tellthemachines I tried testing using Windows NV Access and Mac Voiceover and never had a problem while tabbing to the "Page Parent" combo box. It announced the <select> label as well as the currently selected value. I am still on Mac Catalina as the newer version Mac Big Sur had some rather interesting Voiceover bugs.

This was before testing your pull request.

Thanks.

**Comment by alexstine** [comment]
This was my mistake in testing. Here is my updated view: https://github.com/WordPress/gutenberg/pull/27431#issuecomment-737685857
